### PR TITLE
feat: plugins with a class structure

### PR DESCRIPTION
**Pull request by p-desaintchamas**

### CODE REVIEW

File: recipes/plugins_development.ipynb [hunk 1]
```diff
@@ -0,0 +1,440 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preliminary\n",
+    "\n",
+    "This notebook explains you how to build your 1st plugin. The plugin is a Python script uploaded which will be triggered based on an event you define. For instance you can trigger a plugin each time a labeler click on submit with the `on_submit` handler\n",
+    "The plugin should have separate methods for the different types of events : at the moment these are *on_submit* and *on_review*. These methods will have a predefined set of parameters\n",
```
Comment (pmKili) [inline recipes/plugins_development.ipynb:10]:
```suggestion
    "This notebook will teach you how to build your 1st plugin. A plugin is a Python script uploaded which will be triggered based on an event you will define. For instance, you can trigger a plugin each time a labeler clicks on submit with the `on_submit` handler\n",
    "The plugin should have separate methods for the different types of events: at the moment these are *on_submit* and *on_review*. These methods will have a predefined set of parameters\n",
```

File: recipes/plugins_development.ipynb [hunk 2]
```diff
@@ -0,0 +1,440 @@
...
+    "\n",
+    "    def handle_label(self):\n",
+    "        #custom methods \n",
+    "        print('Custom method handle label')\n",
+    "\n",
+    "    def on_review(self, label: Label, project_id: str, asset_id: str) -> None:\n",
+    "        \"\"\"\n",
+    "        Dedicated handler for Review action \n",
+    "        \"\"\"\n",
+    "        return super().on_review(label, project_id, asset_id)\n",
```
Comment (pmKili) [inline recipes/plugins_development.ipynb:32]: Why would you call the method from PluginCore?

File: recipes/plugins_development.ipynb [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+   ],
+   "source": [
+    "#%pip install kili\n",
+    "\n",
+    "from kili.client import Kili\n",
+    "import os\n",
+    "\n",
+    "api_endpoint = os.getenv('KILI_API_ENDPOINT')\n",
+    "# If you use Kili SaaS, use the url 'https://cloud.kili-technology.com/api/label/v2/graphql'\n",
+    "api_endpoint = 'https://staging.cloud.kili-technology.com/api/label/v2/graphql'\n",
```
Comment (pmKili) [inline recipes/plugins_development.ipynb:78]: Is it a good idea to expose our staging URL publicly?

**Comment by p-desaintchamas** [comment]
Do not delete branch on merge please :)

**Review by gheorghetutunaru — APPROVED** [review]

File: recipes/plugins_development.ipynb [hunk 4]
```diff
@@ -0,0 +1,350 @@
...
+    "\n",
+    "    def on_review(self, label: Label, project_id: str, asset_id: str) -> None:\n",
+    "        \"\"\"\n",
+    "        Dedicated handler for Review action \n",
+    "        \"\"\"\n",
+    "        return super().on_review(label, project_id, asset_id)\n",
+    "\n",
+    "    def on_submit(self, label: Label, project_id: str, asset_id: str) -> None:\n",
+    "        \"\"\"\n",
+    "        Dedicated handler for Review action \n",
```
Comment (gheorghetutunaru) [inline recipes/plugins_development.ipynb:36]: Still Review type

File: recipes/plugins_development.ipynb [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+    "\n",
+    "    def on_review(self, label: Label, asset_id: str) -> None:\n",
+    "        \"\"\"\n",
+    "        Dedicated handler for Review action \n",
+    "        \"\"\"\n",
+    "        return super().on_review(label, asset_id)\n",
+    "\n",
+    "    def on_submit(self, label: Label, asset_id: str) -> None:\n",
+    "        \"\"\"\n",
+    "        Dedicated handler for Review action \n",
```
Comment (gheorghetutunaru) [inline recipes/plugins_development.ipynb:116]: Submit

File: recipes/plugins_development.ipynb [hunk 6]
```diff
@@ -0,0 +1,350 @@
...
+    "class Plugin(PluginCore):\n",
+    "    \"\"\"\n",
+    "    Custom plugin instance\n",
+    "    \"\"\"\n",
+    "\n",
+    "    def handle_label(self):\n",
+    "        #custom methods \n",
+    "        print('Custom method handle label')\n",
+    "\n",
+    "    def on_review(self, label: Label, project_id: str, asset_id: str) -> None:\n",
```
Comment (gheorghetutunaru) [inline recipes/plugins_development.ipynb:28]: I think you forgot to remove project_id here and a few lines below too

**Pull request merged**